src/rosdiscover/observer/observer.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+__all__ = ("Observer",)
+import contextlib
+import os
+from typing import Dict, Iterator
+
+from roswire import App, AppInstance, ROSVersion
+
+from .ros1 import ROS1ObserverConnection
+from .ros2 import ROS2ObserverConnection
+from ..config import Config
+from ..interpreter import NodeContext, SystemSummary
+
+_DEFAULT_URL = os.environ.get("DOCKER_HOST", "unix://var/run/docker.sock")
+
+
+class Observer:
+
+    @classmethod
+    @contextlib.contextmanager
+    def for_container(cls,
+                      container: str,
+                      config: Config,
+                      ) -> Iterator['Observer']:
+        """Constructs and interpreter for a given running container"""
+        app: App = App(config, None)
+        instance = app.attach(container, require_description=False)
+        yield Observer(instance, config)
```

Comment:
Generate appropriate `Observer` implementation based on type of ROS used by App.

Reply:
Fixed